src/streamlink_cli/argparser.py

```diff
@@ -871,6 +871,18 @@ def build_parser():
         Default is 10.0.
         """,
     )
+    transport.add_argument(
+        "--stream-segmented-duration",
+        type=hours_minutes_seconds,
+        metavar="[HH:]MM:SS",
```

Comment:
Is there an easy way to make this accept floats as well?

Reply:
This needs a separate change, because the `hours_minutes_seconds` input parser is used for other arguments and by some plugins as well. I'll have look at this later.
https://github.com/streamlink/streamlink/blame/6.0.1/src/streamlink/utils/times.py#L26-L71

Reply:
Got it, at least whole integers work, which is user-friendly enough.

Reply:
#5495